packages/core/src/saml-applications/routes/utils.ts

```diff
@@ -28,8 +29,366 @@ import {
   samlAttributeNameFormatBasic,
   samlValueXmlnsXsi,
 } from '../libraries/consts.js';
+import { buildSingleSignOnUrl, buildSamlIdentityProviderEntityId } from '../libraries/utils.js';
 import { type SamlApplicationDetails } from '../queries/index.js';
 
+type ValidSamlApplicationDetails = {
+  secret: string;
+  entityId: string;
+  acsUrl: SamlAcsUrl;
+  redirectUri: string;
+  privateKey: string;
+  certificate: string;
+};
+
+export class SamlApplication {
+  public static validateSamlApplicationDetails = (
```

Comment:
How about using the Zod guard instead?

Reply:
We want to throw with detailed error message.